plugins/syncbeaconfollower/plugin.go

```diff
@@ -0,0 +1,194 @@
+	synced  bool
+}
+
+func init() {
+	flag.StringSlice(CfgSyncBeaconFollowNodes, []string{"Gm7W191NDnqyF7KJycZqK7V6ENLwqxTwoKQN4SmpkB24", "9DB3j9cWYSuEEtkvanrzqkzCQMdH1FGv3TawJdVbDxkd"}, "list of trusted nodes to follow their sync status")
+	flag.Int(CfgSyncBeaconMaxTimeWindowSec, 10, "the maximum time window for which a sync payload would be considerable")
+	flag.Int(CfgSyncBeaconMaxTimeOfflineSec, 40, "the maximum time the node should stay without receiving updates")
+	flag.Int(CfgSyncBeaconCleanupInterval, 10, "the interval at which cleanups are done")
+}
+
+var (
+	// plugin is the plugin instance of the sync beacon plugin.
+	plugin         *node.Plugin
+	once           goSync.Once
+	log            *logger.Logger
+	currentBeacons map[string]beaconSync
+	mutex          goSync.RWMutex
+)
+
+var (
+	// ErrMissingFollowNodes is returned if the node starts with no follow nodes list
+	ErrMissingFollowNodes = errors.New("follow nodes list is required")
+)
+
+// Plugin gets the plugin instance.
+func Plugin() *node.Plugin {
+	once.Do(func() {
+		plugin = node.NewPlugin(PluginName, node.Disabled, configure, run)
+	})
+	return plugin
+}
+
+// configure events
+func configure(_ *node.Plugin) {
+	pubKeys := config.Node().GetStringSlice(CfgSyncBeaconFollowNodes)
+
+	if len(pubKeys) == 0 {
+		log.Panicf("Follow node list cannot be empty: %w", ErrMissingFollowNodes)
+	}
+
+	log = logger.NewLogger(PluginName)
+	currentBeacons = make(map[string]beaconSync, len(pubKeys))
+
+	for _, pubKey := range pubKeys {
+		currentBeacons[pubKey] = beaconSync{
+			payload: nil,
+			synced:  false,
+		}
+	}
+
+	messagelayer.Tangle().Events.MessageSolid.Attach(events.NewClosure(func(cachedMessage *message.CachedMessage, cachedMessageMetadata *tangle.CachedMessageMetadata) {
+		cachedMessageMetadata.Release()
+		cachedMessage.Consume(func(msg *message.Message) {
+			messagePayload := msg.Payload()
+			if messagePayload.Type() != syncbeacon.Type {
+				return
+			}
+
+			payload, ok := messagePayload.(*syncbeacon.Payload)
+			if !ok {
+				log.Info("could not cast payload to network delay object")
+				return
+			}
+			handlePayload(payload, msg.IssuerPublicKey())
+		})
+	}))
+}
+
+// handlePayload handles the received payload. It does the following checks:
+// It checks that the issuer of the payload is a followed node.
+// The time that payload was sent is not greater than  CfgSyncBeaconMaxTimeWindowSec. If the duration is longer than CfgSyncBeaconMaxTimeWindowSec, we consider that beacon to be out of sync till we receive a newer payload.
+// More than syncPercentage of followed nodes are also synced, the node is set to synced. Otherwise, its set as desynced.
+func handlePayload(syncBeaconPayload *syncbeacon.Payload, issuerPublicKey ed25519.PublicKey) {
+	mutex.Lock()
+	defer mutex.Unlock()
+
+	issuerPublicKeyStr := issuerPublicKey.String()
+	//check if issuer is in configured beacon follow list
+	if _, ok := currentBeacons[issuerPublicKeyStr]; !ok {
+		return
+	}
```

Comment:
This should be moved out of the `handlePayload` function. 

In configure create an additional map: `publicKey(string)->publicKey([]byte)`
Then you don't need to grab the lock to do this check and save the expensive conversion to `base58` on every beacon message.